src/drive/web/modules/drive/Toolbar/components/CreateOnlyOfficeItem.jsx

```diff
@@ -18,10 +21,13 @@ const CreateOnlyOfficeItem = ({ fileClass }) => {
     [router]
   )
 
-  const handleClick = useCallback(
-    () => router.push(`/onlyoffice/create/${folderId}/${fileClass}`),
-    [router, fileClass, folderId]
-  )
+  const handleClick = useCallback(() => {
+    if (canWriteOfficeDocument()) {
+      router.push(`/onlyoffice/create/${folderId}/${fileClass}`)
+    } else {
+      router.push(`/folder/${folderId}/paywall`)
+    }
+  }, [router, fileClass, folderId])
```

Comment:
canWriteOfficeDocument should be in the dep array, no?

Reply:
I don't think so because canWriteOfficeDocument evaluated each time useEffect render